### PR TITLE
Feature/cc 0044 save court size to db

### PR DESCRIPTION
**Pull request by stevenliu986**

### CODE REVIEW

File: src/resources/court_spec/court_spec.service.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    if (!updatedCourtSpec) {
+      throw new HttpException(`court not found`, 400);
+    }
+    return updatedCourtSpec;
+  }
+
+  async removeCourtSpecById(courtId: string): Promise<{ message: string }> {
+    const isNotDeleted = await this.courtSpecModel.findByIdAndUpdate(courtId, { isDeleted: true });
+    if (!isNotDeleted) {
+      throw new HttpException(`court not found and deletion failed`, 400);
```
Comment (chui-mel) [inline src/resources/court_spec/court_spec.service.ts:54]: same here.

File: data-migrations/migrate-mongo-config.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// In this file you can configure migrate-mongo
+
+const config = {
+  mongodb: {
+    // TODO Change (or review) the url to your MongoDB:
+    url: "mongodb://localhost/court_spec",
```
Comment (GitHubEason) [inline data-migrations/migrate-mongo-config.js:6]: Can we set the URL as an environment variable pls?

File: package.json [hunk 3]
```diff
@@ -59,6 +59,7 @@
     "jest": "28.0.3",
     "lint-staged": "^13.0.1",
     "prettier": "^2.3.2",
+    "prisma": "^3.15.1",
```
Comment (GitHubEason) [inline package.json:62]: Do we still need prisma?

File: src/resources/court_spec/schemas/court_spec_json_example.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (chui-mel) [inline src/resources/court_spec/schemas/court_spec_json_example.txt:1]: this file should name as `court-spec-example.json`

Comment (stkwn) [reply src/resources/court_spec/schemas/court_spec_json_example.txt:1]: update

File: package.json (hunk 3 above)
Comment (stkwn) [reply package.json]: Should we remove it from this card, or open another card to do it?

Comment (chui-mel) [reply package.json]: remove it in this card

Comment (stevenliu986) [reply package.json]: I have already removed it.

File: migrate-mongo-config.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// In this file you can configure migrate-mongo
```
Comment (chui-mel) [inline migrate-mongo-config.js:1]: remove useless comments

File: migrate-mongo-config.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+const config = {
+  mongodb: {
+    // TODO Change (or review) the url to your MongoDB:
+    url: "mongodb://localhost:27017",
```
Comment (chui-mel) [inline migrate-mongo-config.js:6]: use environment variables instead

File: migrations/migrate-mongo-config.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+// In this file you can configure migrate-mongo
```
Comment (chui-mel) [inline migrations/migrate-mongo-config.js:1]: why we need two migrate-mongo-config.js?

File: migrations/migrations/20220624110633-migration-example.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
```
Comment (chui-mel) [inline migrations/migrations/20220624110633-migration-example.js:1]: why the name calls `example`?

**Review by stevenliu986 — COMMENTED** [review]
LGTM

File: courtspecs-migrations/migrate-mongo-config.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+const config = {
+  mongodb: {
+    url: process.env.DATA_MIGRATION_URL,
+
+    databaseName: "courtcanvas",
```
Comment (chui-mel) [inline courtspecs-migrations/migrate-mongo-config.js:5]: this one also needs to put into environment variables

File: src/court_spec/courtSpec.service.ts [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    const court = await this.courtSpecModel.findById(courtId).exec();
+    if (!court || court.isDeleted) {
+      throw new NotFoundException("court not found");
+    }
+    const DeletedCount = await this.courtSpecModel.findByIdAndUpdate(courtId, {
+      isDeleted: true,
+      updatedAt: new Date(),
+    });
+
+    // return { message: `Court ${DeletedCount.name.toUpperCase()} deleted successfully` };
```
Comment (Jzeva) [inline src/court_spec/courtSpec.service.ts:57]: Unused line could be deleted.

File: src/court_spec/schemas/courtSpec-example.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    "centreCircleRadius": 1800,
+    "threePointRadius": 6750,
+    "threePointLine": 900,
+    "lengthOfCorner": 2990,
+    "restrictedAreaLength": 5800,
+    "restrictedAreaWidth": 4900,
+    "sideBorderWidth": 2000,
+    "lineBorderWidth": 50,
+    "description": "This is the default PRO Full Court"
+}
```
Comment (Jzeva) [inline src/court_spec/schemas/courtSpec-example.json:14]: Do we really need this example? I'm just curious. I will add one in my code if we need this.

File: src/main.ts [hunk 12]
```diff
@@ -1,9 +1,11 @@
+import { ValidationPipe } from "@nestjs/common";
 import { NestFactory } from "@nestjs/core";
 import { AppModule } from "./app.module";
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
   app.enableCors();
+  app.useGlobalPipes(new ValidationPipe({ whitelist: true }));
   await app.listen(process.env.PORT || 8080);
 }
```
Comment (Jzeva) [inline src/main.ts:10]: Do you think we need to add a prefix like "v1" for our API router? If you think it is necessary, we can add "v1" as a prefix.

File: src/court_spec/schemas/courtSpec-example.json (hunk 11 above)
Comment (stevenliu986) [reply src/court_spec/schemas/courtSpec-example.json:14]: I think it is for reference.

**Review by Jzeva — APPROVED** [review]

File: courtspecs-migrations/migrate-mongo-config.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
+const config = {
+  mongodb: {
+    url: process.env.DATA_MIGRATION_URL,
+
+    databaseName: process.env.DATABASE_NAME,
```
Comment (GitHubEason) [inline courtspecs-migrations/migrate-mongo-config.js:5]: when I run this script,  process.env.DATA_MIGRATION_URL, and process.env.DATABASE_NAME  are undefined.

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request merged**